src/Hubbup.Web/Controllers/MikLabelerController.cs

```diff
@@ -42,29 +44,46 @@ public async Task<IActionResult> Index(string repoSet)
             //This line re-trains the ML Model
             //MLHelper.BuildAndTrainModel(_hostingEnvironment.ContentRootPath + "/ML/issueData.tsv", _hostingEnvironment.ContentRootPath + ModelPath, MyTrainerStrategy.OVAAveragedPerceptronTrainer);
 
-            var labeler = new Labeler(_hostingEnvironment.ContentRootPath + ModelPath);
-            var issues = await gitHub.Issue.GetAllForRepository("aspnet", "AspNetCore", new ApiOptions { PageSize = 100, PageCount = 1 });
+            var existingAreaLabels =
+                (await gitHub.Issue.Labels.GetAllForRepository("aspnet", "AspNetCore"))
+                .Where(label => label.Name.StartsWith("area-", StringComparison.OrdinalIgnoreCase))
+                .ToList();
 
-            var issueList = new List<Issue>();
-            var predictionLabels = new List<GitHubIssuePrediction>();
-            foreach (Issue issue in issues)
+            var excludeAllAreaLabelsQuery =
+                string.Join(
+                    " ",
+                    existingAreaLabels.Select(label => $"-label:\"{label.Name}\""));
+
+            var getIssuesRequest = new SearchIssuesRequest(excludeAllAreaLabelsQuery)
             {
-                if (issue.PullRequest != null)
+                Is = new[] { IssueIsQualifier.Issue, IssueIsQualifier.Open },
+                Repos = new RepositoryCollection
                 {
-                    continue;
-                }
+                    { "aspnet", "AspNetCore" }
+                },
+            };
+
+            var issueSearchResult = await gitHub.Search.SearchIssues(getIssuesRequest);
```

Comment:
Instead of using `gitHub.Issue.GetAllForRepository` I decided to use `gitHub.Search.SearchIssues`, which allows excluding issues by label. This is nice because with the old logic it would get, say, 100 issues, and then filter out all but 6-7 because the rest were labelled. Now it gets the the top, say, 100 issues, all of which don't have labels.

Reply:
Okay awesome. Didn't know about this API. Definitely a better way to approach this 👍

Reply:
Yeah I never used it before because it provides incomplete Issue information. But it's enough information for the prediction: title, full body, and some other metadata. It wasn't good enough for things like the main part of Hubbup because it had to do multi-repo queries and whatnot.